framework/include/utils/MathUtils.h

```diff
@@ -30,22 +28,51 @@ Real poly2Log(Real x, Real tol, unsigned int derivative_order);
 Real poly3Log(Real x, Real tol, unsigned int derivative_order);
 Real poly4Log(Real x, Real tol, unsigned int derivative_order);
 Real taylorLog(Real x);
-Point barycentricToCartesian2D(const Point & p0,
-                               const Point & p1,
-                               const Point & p2,
-                               const Real b0,
-                               const Real b1,
-                               const Real b2);
-Point barycentricToCartesian3D(const Point & p0,
-                               const Point & p1,
-                               const Point & p2,
-                               const Point & p3,
-                               const Real b0,
-                               const Real b1,
-                               const Real b2,
-                               const Real b3);
-Point circumcenter2D(const Point & p0, const Point & p1, const Point & p2);
-Point circumcenter3D(const Point & p0, const Point & p1, const Point & p2, const Point & p3);
+/**
+ * Evaluate Cartesian coordinates of any center point of a triangle given Barycentric
+ * coordinates of center point and Cartesian coordinates of triangle's vertices
+ * @param p0,p1,p2 are the three non-collinear vertices in Cartesian coordinates
+ * @param b0,b1,b2 is the center point in barycentric coordinates, e.g. (1/3,1/3,1/3) for a centriod
+ * @return the center point of triangle in Cartesian coordinates
+ */
+libMesh::Point barycentricToCartesian2D(const libMesh::Point & p0,
+                                        const libMesh::Point & p1,
+                                        const libMesh::Point & p2,
+                                        const Real b0,
+                                        const Real b1,
+                                        const Real b2);
+/**
+ * Evaluate Cartesian coordinates of any center point of a tetrahedron given Barycentric
+ * coordinates of center point and Cartesian coordinates of tetrahedon's vertices
+ * @param p0,p1,p2,p3 are the three non-coplanar vertices in Cartesian coordinates
+ * @param b0,b1,b2,b3 is the center point in barycentric coordinates, e.g. (1/4,1/4,1/4,1/4) for a
+ * centriod
+ * @return the center point of tetrahedron in Cartesian coordinates
+ */
+libMesh::Point barycentricToCartesian3D(const libMesh::Point & p0,
+                                        const libMesh::Point & p1,
+                                        const libMesh::Point & p2,
+                                        const libMesh::Point & p3,
+                                        const Real b0,
+                                        const Real b1,
+                                        const Real b2,
+                                        const Real b3);
```

Comment:
should they sum to 1?

Reply:
That's correct, there's one less dof in the barycentric coordinates. Maybe I should reduce the number of params to be passed in or have a check to make sure they sum to one?

Reply:
you can add an assert that checks that they sum to 1

Reply:
Added `mooseAssert`